components/authz-service/storage/v2/project_rules_status.go

```diff
@@ -0,0 +1,37 @@
+package v2
+
+import (
+	"fmt"
+)
+
+// ProjectRulesStatus is an enum that represents the states
+// a project's rules can be in.
+type ProjectRulesStatus int
+
+const (
+	// The project has rules but all are applied
+	Applied ProjectRulesStatus = iota
+	// The project has rules and at least one is staged
+	EditsPending
+	// The project has no staged or applied rules
+	NoRules
+	// RulesStatusError occurs in some error situation
+	RulesStatusError
```

Comment:
[nit] If we changed the order here to have `RulesStatusError` be _first_, we could just `return nil, 0, err` in the postgres code. But this is fine, too.

Reply:
I guess that's technically true. Mostly makes the tests easier to read on failure. Otherwise, I don't think you'd really see the enum in it's integer context. Easy enough to fix though.